pkg/workspace/templates.go

```diff
@@ -217,6 +219,15 @@ func retrievePulumiTemplates(templateName string, offline bool) (TemplateReposit
 	subDir := templateDir
 	if templateName != "" {
 		subDir = filepath.Join(subDir, templateName)
+
+		// Provide a nicer error message when the template can't be found (dir doesn't exist).
+		_, err := os.Stat(subDir)
```

Comment:
This does suffer from TOCTOU issues. I wonder, is it possible instead to check the error when we actually try to open the file?

Reply:
If this can be moved to the time of open, great - but since this is an operation inside a folder we own all writes/reads on, and only impacts the ability to provide a great error message, I'm not too worried about about the timing issue here.

Reply:
Unfortunately, based on how the code's structured, it's not as easy as simply moving it to the time of read, so I'm going to merge as-is.